app/main.py

```diff
@@ -1 +1,30 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        Animal.alive.append(self)
+        self.hidden = False
+        self.health = health
+
+    def __sub__(self, other: int) -> int:
+        return self.health - other
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(prey: Animal) -> None:
+        if not prey.hidden and isinstance(prey, Herbivore):
+            prey.health -= 50
+        if prey.health <= 0:
+            Animal.alive.remove(prey)
```

Comment:
The `bite` method in the `Carnivore` class checks if the prey is a `Herbivore` and not hidden before reducing its health. However, the method does not handle the case where the prey's health becomes zero or negative after the bite. It correctly removes the prey from the `alive` list if its health is zero or less, but it might be beneficial to ensure that the prey's health does not go below zero to avoid negative health values.